ion-schema/src/isl/ranges.rs

```diff
@@ -132,7 +132,51 @@ pub type TimestampRange = base::Range<Timestamp>;
 impl RangeValidation<Timestamp> for TimestampRange {}
 
 pub type TimestampPrecisionRange = base::Range<TimestampPrecision>;
-impl RangeValidation<TimestampPrecision> for TimestampPrecisionRange {}
+impl RangeValidation<TimestampPrecision> for TimestampPrecisionRange {
+    fn is_empty(start: &Limit<TimestampPrecision>, end: &Limit<TimestampPrecision>) -> bool {
+        use crate::isl::util::TimestampPrecision::*;
+
+        match (start, end) {
+            (Limit::Inclusive(lower), Limit::Inclusive(upper)) => lower > upper,
+            (Limit::Exclusive(lower), Limit::Inclusive(upper))
+            | (Limit::Inclusive(lower), Limit::Exclusive(upper)) => lower >= upper,
+            (Limit::Exclusive(lower), Limit::Exclusive(upper)) => {
+                let start_value = match lower {
+                    Year => -4,
+                    Month => -3,
+                    Day => -2,
+                    Minute => -1,
+                    Second => 0,
+                    Millisecond => 3,
+                    Microsecond => 6,
+                    Nanosecond => 9,
+                    OtherFractionalSeconds(scale) => *scale,
+                };
+
+                let end_value = match upper {
+                    Year => -4,
+                    Month => -3,
+                    Day => -2,
+                    Minute => -1,
+                    Second => 0,
+                    Millisecond => 3,
+                    Microsecond => 6,
+                    Nanosecond => 9,
+                    OtherFractionalSeconds(scale) => *scale,
+                };
+
+                // Checking for e.g. range::[exclusive::1, exclusive::2] which is empty.
+                let adjusted_lower = start_value.checked_add(1);
+                // If the _lower_ bound wraps around when we add one, then we know it's empty.
+                if adjusted_lower.is_none() {
+                    return true;
+                }
```

Comment:
I'm not sure I understand this. Why would the lower bound wrap around by adding one? The lower bound cannot wrap around unless it's already the _max_ value, in which case we can detect that it's empty by `lower >= upper`.

Reply:
Thats right, thanks for catching it. I forgot about the `Limit::Unbound` enum variant you added recently.
Since this isn't really a max value doesn't require to check for `is_none` here. Maybe can just be `start_value + 1` in that case.